src/main/resources/org/jenkinsci/plugins/badge/BadgeAction/index.groovy

```diff
@@ -30,44 +30,79 @@ l.layout {
 </style>
 """)
 
-        def base =  "${app.rootUrl}${my.project.url}";
-        def badge = base + "badge/icon"
-        
         def fullJobName = h.escape(my.project.fullName);
-        def publicbadge = "${app.rootUrl}buildStatus/icon?job=${fullJobName}";
+        def jobUrlWithView =  "${app.rootUrl}${my.project.url}";
+        def jobUrlWithoutView =  "${app.rootUrl}job/${fullJobName}";
```

Comment:
This is not correct as soon as `project.fullName != project.name`. Try e.g. with Cloudbees Folder Plugin. Workaround here: https://github.com/daniel-beck/shareable-link-plugin